src/plonkish/backend/halo2.rs

```diff
@@ -496,17 +549,76 @@ pub fn halo2_verify(
 
 /// Halo2 setup
 pub struct Setup {
+    pub k: u32,
```

Comment:
The fields of `Setup` don't have to be public as you have exposed them through the getters.